HARK/core.py

```diff
@@ -1058,20 +1073,15 @@ def mill(self):
         none
         '''
         # Make a dictionary of inputs for the millRule
-        reap_vars_string = ''
-        for name in self.reap_vars:
-            reap_vars_string += ' \'' + name + '\' : self.' + name + ','
-        const_vars_string = ''
-        for name in self.const_vars:
-            const_vars_string += ' \'' + name + '\' : self.' + name + ','
-        mill_dict = eval('{' + reap_vars_string + const_vars_string + '}')
+        mill_dict = copy(self.reap_vars)
+        mill_dict.update(self.const_vars)
 
         # Run the millRule and store its output in self
         product = self.millRule(**mill_dict)
-        for j in range(len(self.sow_vars)):
-            this_var = self.sow_vars[j]
-            this_product = getattr(product, this_var)
-            setattr(self, this_var, this_product)
+
+        for sow_var in self.sow_state:
+            this_product = getattr(product, sow_var)
```

Comment:
This simplifies if `millRule` returns a dict-like: `self.sow_state.update(product)`

Reply:
chose for `millRule` to return a tuple

Reply:
(closer to Dolo, supposed to be good for numba)